test/tools/test_tools_time.py

```diff
@@ -2,13 +2,39 @@
 from __future__ import annotations
 
 import datetime
+from typing import TYPE_CHECKING
 
 import pytest
 import pytz
 
+from sopel.db import SopelDB
 from sopel.tools import time
 
 
+if TYPE_CHECKING:
+    from sopel.config import Config
+
+
+TMP_CONFIG = """
+[core]
+owner = Pepperpots
+db_filename = {db_filename}
+"""
+
+
+@pytest.fixture
+def tmpconfig(configfactory, tmpdir):
+    content = TMP_CONFIG.format(db_filename=tmpdir.join('test.sqlite'))
+    return configfactory('default.cfg', content)
+
+
+@pytest.fixture
+def db(tmpconfig):
+    db = SopelDB(tmpconfig)
+    # TODO add tests to ensure db creation works properly, too.
```

Comment:
If this applies to uses of the DB in other tests, it should be a follow-up issue, not a TODO left here (in a place where it's likely to get forgotten).